nucypher/utilities/sandbox/blockchain.py

```diff
@@ -0,0 +1,131 @@
+from typing import List
+
+from umbral.keys import UmbralPrivateKey
+from web3.middleware import geth_poa_middleware
+
+from nucypher.blockchain.eth import constants
+from nucypher.blockchain.eth.chains import Blockchain
+from nucypher.utilities.sandbox.constants import DEVELOPMENT_ETH_AIRDROP_AMOUNT, \
+    DEFAULT_NUMBER_OF_URSULAS_IN_DEVELOPMENT_NETWORK, TEST_URSULA_INSECURE_DEVELOPMENT_PASSWORD
+
+
+def token_airdrop(token_agent, amount: int, origin: str, addresses: List[str]):
+    """Airdrops tokens from creator address to all other addresses!"""
+
+    def txs():
+        for address in addresses:
+            txhash = token_agent.contract.functions.transfer(address, amount).transact({'from': origin})
+            yield txhash
+
+    receipts = list()
+    for tx in txs():  # One at a time
+        receipt = token_agent.blockchain.wait_for_receipt(tx)
+        receipts.append(receipt)
+    return receipts
+
+
+class TesterBlockchain(Blockchain):
+    """
+    Blockchain subclass with additional test utility methods and options.
+    """
+
+    _instance = None
+    _default_network = 'tester'
+    _test_account_cache = list()
+
+    def __init__(self, test_accounts=None, poa=True, airdrop=True, *args, **kwargs):
+
+        # Depends on circumflex
+        super().__init__(*args, **kwargs)
+
+        # For use with Proof-Of-Authority test-blockchains
+        if poa is True:
+            w3 = self.interface.w3
```

Comment:
In recent web3, one can do `w3 = web3.auto.gethdev.w3` (although, manually injecting into `middleware_stack` should also continue to work... Until high velocity breaks it again!)

Reply:
This is only for supporting test runs with geth... at the moment - Ill look at the `web3.auto.gethdev.w3` API.
```
        # For use with Proof-Of-Authority test-blockchains
        if poa is True:
            w3 = self.interface.w3
            w3.middleware_stack.inject(geth_poa_middleware, layer=0)
```